src/network/C4Network2IO.cpp

```diff
@@ -460,28 +460,49 @@ bool C4Network2IO::InitPuncher(C4NetIO::addr_t nPuncherAddr)
 	if (!pNetIO_UDP)
 		return false;
 	// save address
-	PuncherAddr = nPuncherAddr;
+	switch (nPuncherAddr.GetFamily())
+	{
+	case C4NetIO::HostAddress::IPv4:
+	    PuncherAddrIPv4 = nPuncherAddr;
+	    break;
+	case C4NetIO::HostAddress::IPv6:
+	    PuncherAddrIPv6 = nPuncherAddr;
+	    break;
+	default:
```

Comment:
Maybe replace this with `case C4NetIO::HostAddress::Unknown: assert(!"insert meaningful error here");` so in case the IETF specifies IPv8 at some point at least we'll have a non-exhaustion warning that we can ignore